File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> store;
        for (int i = 0; i < nums.size(); i++) {
            int left = target - nums[i];
            if (store.find(left) != store.end()) {
                return {store[left], i};
            }
            store[nums[i]] = i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        storage = {}      
        for i in range(len(nums)):
            left = target - nums[i]      
            if left in storage:
                return [storage[left], i]   
            storage[nums[i]] = i  
        return []